TShockAPI/GetDataHandlers.cs

```diff
@@ -3720,7 +3720,7 @@ private static bool HandleSpawnBoss(GetDataHandlerArgs args)
 					break;
 			}
 
-			if (NPCID.Sets.MPAllowedEnemies[thingType])
+			if (!invasions.Contains(thingType) && NPCID.Sets.MPAllowedEnemies[thingType])
```

Comment:
The logic looks weird. If `MPAllowedEnemies[thingType]` is true, thing must be a positive number. `invasions` are negatives so definitely they can't contains a positive number.

```suggestion
			if (invasions.Contains(thingType) || NPCID.Sets.MPAllowedEnemies[thingType])
```

Reply:
Well, before we were logging every npc that was summoned for some reason (Stuff like "the bunny", etc which arent spawnable but alright). I thought this check would prevent spam from those running a server with anonymous boss invasions turned off, however, it turns out this has caused more harm than good.